src/MongoDB/Driver/Manager.h

```diff
@@ -47,13 +48,11 @@ class MongoDBDriverManagerData
 			throw MongoDriver::Utils::throwRunTimeException("Serialization of MongoDB\\Driver\\Manager is not allowed");
 		}
 
-		void sweep() {
-			mongoc_client_destroy(m_client);
```

Comment:
Just to confirm, you're relying on the client's memory to get freed when the HHVM worker thread dies?